File: ApiBaseCore/ApiBaseCore.Application/Interfaces/IUserService.cs
using ApiBaseCore.Application.DTOs;

namespace ApiBaseCore.Application.Interfaces;

public interface IUserService
{
    Task<List<PerfilUsuarioDto>> GetUsersAsync(UserFilterDto filterDto);
    Task<PerfilUsuarioDto> GetUserByIdAsync(Guid id);
    Task<PerfilUsuarioDto> CreateUserAsync(RegistroUsuarioDto createUserDto);
    Task<PerfilUsuarioDto> UpdateUserAsync(Guid id, UpdateUserDto updateDto);
    Task DeleteUserAsync(Guid id);
}

File: ApiBaseCoreJava/src/main/java/com/polancou/apibasecore/application/interfaces/IUserService.java
package com.polancou.apibasecore.application.interfaces;

import com.polancou.apibasecore.application.dtos.PerfilUsuarioDto;
import com.polancou.apibasecore.application.dtos.RegistroUsuarioDto;
import com.polancou.apibasecore.application.dtos.UpdateUserDto;
import com.polancou.apibasecore.application.dtos.UserFilterDto;

import java.util.List;
import java.util.UUID;

public interface IUserService {
    List<PerfilUsuarioDto> getUsers(UserFilterDto filterDto);
    PerfilUsuarioDto getUserById(UUID id);
    PerfilUsuarioDto createUser(RegistroUsuarioDto createUserDto);
    PerfilUsuarioDto updateUser(UUID id, UpdateUserDto updateDto);
    void deleteUser(UUID id);
}
